CMakeLists.txt

```
@@ -1,30 +1,66 @@
-project(xunused)
-
-# Let find_package(...) search for the latest version when multiple versions of
-# the same library are available. Therefore ...
-#
-#   1) set natural sort order to compare contiguous digits as whole numbers, and
-#   2) do ordering in descending mode to test latest found folder first.
-#
-set(CMAKE_FIND_PACKAGE_SORT_ORDER NATURAL)
-set(CMAKE_FIND_PACKAGE_SORT_DIRECTION DEC)
-
-find_package(LLVM  ${LLVM_VERSION_REQUIRED} CONFIG REQUIRED)
-message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
-message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
-
-find_package(Python3 COMPONENTS Interpreter)
-
-if (LLVM_VERSION_MAJOR GREATER_EQUAL 16)
-    if (LLVM_VERSION_MAJOR GREATER_EQUAL 18)
-        find_package(Clang ${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR} CONFIG REQUIRED)
-    else()
-        find_package(Clang ${LLVM_VERSION_MAJOR} CONFIG REQUIRED)
-    endif()
-    message(STATUS "Using ClangConfig.cmake in: ${Clang_DIR}")
-endif (LLVM_VERSION_MAJOR GREATER_EQUAL 16)
 
+if (CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+    set(xunused_standalone_build ON)
+    message(STATUS "xunused: standalone build")
+else()
+    set(xunused_standalone_build OFF)
+    message(STATUS "xunused: build as part of LLVM")
+endif()
+
+if (xunused_standalone_build)
+    project(xunused)
+
+    # Let find_package(...) search for the latest version when multiple versions of
+    # the same library are available. Therefore ...
+    #
+    #   1) set natural sort order to compare contiguous digits as whole numbers, and
+    #   2) do ordering in descending mode to test latest found folder first.
+    #
+    set(CMAKE_FIND_PACKAGE_SORT_ORDER NATURAL)
+    set(CMAKE_FIND_PACKAGE_SORT_DIRECTION DEC)
+
+    find_package(LLVM ${LLVM_VERSION_REQUIRED} CONFIG REQUIRED)
+    message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
+    message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
+
+    if (LLVM_VERSION_MAJOR GREATER_EQUAL 16)
+        if (LLVM_VERSION_MAJOR GREATER_EQUAL 18)
+            find_package(Clang ${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR} CONFIG REQUIRED)
+        else()
+            find_package(Clang ${LLVM_VERSION_MAJOR} CONFIG REQUIRED)
+        endif()
+        message(STATUS "Using ClangConfig.cmake in: ${Clang_DIR}")
+    endif (LLVM_VERSION_MAJOR GREATER_EQUAL 16)
+
+    set(xunused_llvm_dir ${LLVM_DIR})
+    set(xunused_include_dirs
+        ${LLVM_INCLUDE_DIRS}
+        ${CLANG_INCLUDE_DIRS}
+    )
+else()
+    set(xunused_llvm_dir ${CMAKE_SOURCE_DIR})
+    set(xunused_include_dirs
+        ${LLVM_SOURCE_DIR}/include
+        ${LLVM_EXTERNAL_CLANG_SOURCE_DIR}/include
+        ${LLVM_BINARY_DIR}/include
+        ${LLVM_BINARY_DIR}/tools/clang/include
+    )
+
+    if (UNIX)
+        # When building as part of LLVM on Unix, LLVM and Clang are built with
+        # -fno-rtti, so we need to disable RTTI for xunused as well.
+        # https://github.com/mgehre/xunused/issues/1
+        add_compile_options(-fno-rtti)
```

RTTI is optional in LLVM and off by default. You can handle both cases with
```suggestion
        if(NOT LLVM_ENABLE_RTTI)
          add_compile_options(-fno-rtti)
        endif()
```